src/plugins/liveobjects/liveobject.ts

```diff
@@ -41,10 +90,14 @@ export abstract class LiveObject<T extends LiveObjectData = LiveObjectData> {
   }
 
   /**
+   * Sets a new data reference for the LiveObject and returns an update object that describes the changes applied based on the object's previous value.
+   *
    * @internal
    */
-  setData(newDataRef: T): void {
+  setData(newDataRef: TData): TUpdate {
+    const update = this._updateFromDataDiff(newDataRef);
     this._dataRef = newDataRef;
+    return update;
```

Comment:
very minor, but
```suggestion
    this._dataRef = newDataRef;
    return this._updateFromDataDiff(newDataRef);
```

Reply:
the order was intentional - `_updateFromDataDiff` uses the current data from `this._dataRef`, so you are supposed to call it before changin the `this._dataRef`.
however, I can see how this is easily missed and error-prone. I will change it to explicitly expect a `currentDataRef` parameter, so it is more intuitive how you need to call it

Reply:
updated in https://github.com/ably/ably-js/compare/cf43cb204060344b6606f7c180d41ff5681b32a6..4bca1c797caa207de96f3ff767d916f2426489b8

Reply:
looks better now, thanks!